src/csharp/experimental/build_native_ext_for_android.sh

```diff
@@ -23,17 +23,28 @@ mkdir -p build
 cd build
 
 # set to the location where Android SDK is installed
-# e.g. ANDROID_NDK_PATH="$HOME/android-ndk-r16b"
+ANDROID_SDK_PATH="$HOME/Android/Sdk"
 
-cmake ../.. \
-  -DCMAKE_SYSTEM_NAME=Android \
-  -DCMAKE_SYSTEM_VERSION=15 \
-  -DCMAKE_ANDROID_ARCH_ABI=armeabi-v7a \
+# set to location where Android NDK is installed, usually a subfolder of Android SDK
+# install the Android NDK through the Android SDK Manager
+ANDROID_NDK_PATH=${ANDROID_SDK_PATH}/ndk-bundle
+
+# set to location of the cmake executable
+# by default, use cmake binary from the Android SDK
+CMAKE_PATH="${ANDROID_SDK_PATH}/cmake/3.6.4111459/bin/cmake"
+
+# ANDROID_ABI in ('arm64-v8a', 'armeabi-v7a')
+${CMAKE_PATH} ../.. \
+  -DCMAKE_TOOLCHAIN_FILE="${ANDROID_NDK_PATH}/build/cmake/android.toolchain.cmake" \
   -DCMAKE_ANDROID_NDK="${ANDROID_NDK_PATH}" \
   -DCMAKE_ANDROID_STL_TYPE=c++_static \
   -DRUN_HAVE_POSIX_REGEX=0 \
   -DRUN_HAVE_STD_REGEX=0 \
   -DRUN_HAVE_STEADY_CLOCK=0 \
-  -DCMAKE_BUILD_TYPE=Release
+  -DCMAKE_BUILD_TYPE=Release \
+  -DANDROID_PLATFORM=android-28 \
```

Comment:
can you comment on why android-28 is the right choice here?

Reply:
Based on https://developer.android.com/ndk/guides/cmake and https://developer.android.com/ndk/guides/stable_apis  setting `ANDROID_PLATFORM=android-28`  seems way too ambitious. We should aim at much older version if we are going to distribute the libgrpc_csharp_ext.so files as part of the nuget.

Reply:
Based on android version usage here https://developer.android.com/about/dashboards/, it seems a good starting point would be to support 4.4+(Kitkat) and ANDROID_PLATFORM=android-19.

Reply:
Yes, agree.

Reply:
Please note you need the `arm64-v8a` build for 64-bit environments, the `armeabi-v7a` won't work.

Reply:
Also, I'm not sure the the C++ wrappers are needed (`CMAKE_ANDROID_STL_TYPE=c++_static`), the plain C entry points for the P/Invoke mechanism should suffice, can we use a value of `none` here? See https://cmake.org/cmake/help/v3.7/variable/CMAKE_ANDROID_STL_TYPE.html.